### PR TITLE
HOSTEDCP-2067: [CPO Refactor] Refactor cloud-controller-manager

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/v2/kas/deployment.go [hunk 1]
```diff
@@ -7,7 +7,6 @@ import (
 	"strings"
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/v2/kas/deployment.go:7]: can we provide some context con the commit desc on why we're dropping this?

**Comment by muraee** [comment]
/override "Red Hat Konflux / hypershift-operator-main-enterprise-contract / hypershift-operator-main"

**Pull request merged**